src/main/java/com/atlassian/db/replica/api/context/Reason.java

```diff
@@ -0,0 +1,60 @@
+package com.atlassian.db.replica.api.context;
+
+import java.util.Objects;
+
+/**
+ * Describes the reason to choose either replica or main database connection.
+ */
+public final class Reason {
+    private final String name;
+    private final boolean isRunOnMain;
+
+    private Reason(final String name, boolean isRunOnMain) {
+        this.name = name;
+        this.isRunOnMain = isRunOnMain;
+    }
+
+    public static final Reason API_CALL = new Reason("API_CALL", true);
+    public static final Reason REPLICA_INCONSISTENT = new Reason("REPLICA_INCONSISTENT", true);
+    public static final Reason READ_OPERATION = new Reason("READ_OPERATION", false);
+    public static final Reason WRITE_OPERATION = new Reason("WRITE_OPERATION", true);
+    public static final Reason LOCK = new Reason("LOCK", true);
+    public static final Reason MAIN_CONNECTION_REUSE = new Reason(
+        "MAIN_CONNECTION_REUSE",
+        true
+    ); // TODO: Add cause to this reason!! (an sql/reason that is responsible for the switch)
```

Comment:
- [ ] TODO: Add cause to this reason!! (an sql/reason that is responsible for the switch)

Reply:
There's already a task for it